client/components/bookClubPage/BookClubItem.jsx

```diff
@@ -1,29 +1,25 @@
-import React, { useState } from 'react';
-import { Link } from 'react-router-dom';
+import React, { useContext } from 'react';
 import { Container, Image } from 'react-bootstrap';
-import default_bookclub from '../../assets/images/bookclubs_sample/bookclub_6.svg';
+import { AppContext } from '../../context/context.jsx';
 
-const BookClubItem = ({ id, title, desc, splash, imgVariant }) => {
-  const [selected, setSelected] = useState(false);
+const BookClubItem = ({ current }) => {
+  const { club, getClubById } = useContext(AppContext);
+
+  const handleClicked = () => getClubById(current._id);
 
   return (
-    <Link to={`/clubs/${id}`}>
-      <Container className='my-1 p-1'>
-        {/* <Card.ImgOverlay>
+    <Container className='my-1 p-1' onClick={handleClicked}>
+      {/* <Card.ImgOverlay>
```

Comment:
Should the click event be on the container or the image? When I click anywhere on the carousel I get 404 errors for trying to get /bookclubs/undefined

Reply:
change request handled